### PR TITLE
Added test for invariantHolds and IsApplicable for shoot or pass play

### LINKED ISSUE #2260: Implement ShootOrPassPlay isApplicable and invariantHolds tests
### Description of the task

`isApplicable()` determines if the play should start running
`invariantHolds()` determines if the play should continue running (`isApplicable()` was true earlier to start the play)

Currently, the `isApplicable()` and `invariantHolds()` functions of the play are untested. We had a lot of bugs because of this at Virtual Robocup. We need to implement a test for each.

Take a look at an example of how to do this for `HaltPlay`, the setup of the Play, World, GameState and ASSERT_TRUE/FALSE will be similar for your test: https://github.com/UBC-Thunderbots/Software/pull/2251/files

NOTE: 
* That PR has been heavily commented to make it easier to follow along. You only need to add a couple comments to clarify what your testing, nothing more.
* The createGameState function has been implemented to conviently create GameState from referee commands. You need to poke around `software/world/game_state.cpp` to figure out the right sequence of RefereeCommands. (You should already be slightly familiar with this file from the interview :))

### Acceptance criteria

- [ ] Make sure `isApplicable` and `invariantHolds` are correct: https://robocup-ssl.github.io/ssl-rules/sslrules.html (Rule Book)
- [ ] Implement tests for `isApplicable` and `invariantHolds`

### Blocked By
N/A

### PR DESCRIPTION
**Pull request by Bvasilchikov**
### Description

this PR implements tests for `invariantHolds` and `isApplicable` in `ShootOrPassPlay`

### Testing Done

 `bazel build //software/ai/hl/stp/play:shoot_or_pass_play_test`
 `bazel test //software/ai/hl/stp/play:shoot_or_pass_play_test`

### Resolved Issues

resolves #2260

### Length Justification

n/a

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — APPROVED** [review]
lgtm!

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
ah woops just realized CI is failing, make sure to run `formatting_scripts/fix_formatting.sh`

File: src/software/ai/hl/stp/play/shoot_or_pass_play_test.cpp [hunk 1]
```diff
@@ -91,3 +91,32 @@ TEST_F(ShootOrPassPlayTest, test_shoot_or_pass_play_with_keep_away)
...
+    // pass it in.
+    auto play_config = std::make_shared<ThunderbotsConfig>()->getPlayConfig();
+
+    // World: A blank testing world we will manipulate for the test
+    auto world = ::TestUtil::createBlankTestingWorld();
+
+    // ShootOrPassPlay: The play under test
+    auto shoot_or_pass_play = ShootOrPassPlay(play_config);
+    world.updateGameState(
+        ::TestUtil::createGameState(RefereeCommand::FORCE_START, RefereeCommand::HALT));
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/shoot_or_pass_play_test.cpp:110]: can we also have a test for one of the special game states like PREPARE_KICKOFF_US

**Review by jonathanlew — COMMENTED** [review]
just one non-blocking comment

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by naraiamsakun — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
lgtm!

**Pull request merged**